File: dart/raindrops/lib/raindrops.dart
class Raindrops {
  String convert(num n) {
    String res = '';
    Map<num, String> map = { 3: 'Pling', 5: 'Plang', 7: 'Plong'};
    map.forEach((k, v) => n % k == 0 ? res += v : null);
    return res != '' ? res : n.toString();
  }
}

File: kotlin/raindrops/src/main/kotlin/Raindrops.kt
package Raindrops

var map = linkedMapOf(3 to "Pling", 5 to "Plang", 7 to "Plong")

fun convert(n: Int): String {
  var res = StringBuilder();

  for ((num, sound) in map.entries) {
    if (n % num == 0) {
      res.append(sound);
    }
  }

  return if (res.isEmpty()) n.toString() else res.toString()
}